### ISSUE TITLE
Emmet support for JSX

### ISSUE THREAD

**Issue by l2dy** [rescript-lang/rescript-vscode#286]
Is it possible to get Emmet support in ReScript JSX? https://code.visualstudio.com/docs/editor/emmet

**Comment by cristianoc** [comment]
I think @chenglou added some snippets -- not sure if using Emmet.

**Comment by amiralies** [comment]
This is not related to rescript-vscode.
You have to enable emmet for rescript file type.
take a look at [here](https://code.visualstudio.com/docs/editor/emmet#_emmet-abbreviations-in-other-file-types).
based on that you have to put something like this in your config file.
```json
"emmet.includeLanguages": {
    "rescript": "javascriptreact"
}
```

**Comment by l2dy** [comment]
> Emmet has no knowledge of these new languages, and so there might be Emmet suggestions showing up in non HTML/CSS contexts.

Is it possible to optimize the experience? Emmet completions are popping out wherever I type if I add `rescript` to `emmet.includeLanguages`.

**Comment by amiralies** [comment]
I'm not sure if we can improve this from rescript side. Are you aware of other language plugins doing this?

**Comment by l2dy** [comment]
> I'm not sure if we can improve this from rescript side. Are you aware of other language plugins doing this?

Yes, [Svelte](https://marketplace.visualstudio.com/items?itemName=svelte.svelte-vscode) plugin supports Emmet through [vscode-emmet-helper](https://www.npmjs.com/package/vscode-emmet-helper).
